tests/test_metadata.py

```diff
@@ -247,3 +243,227 @@ def test_complete(self):
         ]
         assert raw["dynamic"] == ["Obsoletes-Dist"]
         assert raw["description"] == "This description intentionally left blank.\n"
+
+
+class TestMetadata:
+    def test_from_email(self):
+        metadata_version = "2.3"
+        meta = metadata.Metadata.from_email(f"Metadata-Version: {metadata_version}")
+
+        assert meta.metadata_version == metadata_version
+
+    @pytest.mark.parametrize(
+        "attribute",
+        [
+            "description",
+            "home_page",
+            "download_url",
+            "author",
+            "author_email",
+            "maintainer",
+            "maintainer_email",
+            "license",
+        ],
+    )
+    def test_single_value_unvalidated_attribute(self, attribute):
+        value = "Not important"
+        meta = metadata.Metadata.from_raw({attribute: value})
+
+        assert getattr(meta, attribute) == value
+
+    @pytest.mark.parametrize(
+        "attribute",
+        [
+            "supported_platforms",
+            "platforms",
+            "classifiers",
+            "provides_dist",
+            "obsoletes_dist",
+        ],
+    )
+    def test_multi_value_unvalidated_attribute(self, attribute):
+        values = ["Not important", "Still not important"]
+        meta = metadata.Metadata.from_raw({attribute: values})
+
+        assert getattr(meta, attribute) == values
+
+    @pytest.mark.parametrize(
```

Comment:
It's definitely an open question. I'm personally fine with being strict and saying folks are expected to use raw metadata parsing in those instances.

Reply:
We can always start with it being invalid and loosen it later if it turns out there is a lot of breakage.

I have vague memories that some versions of wheel was emitting it though, so it may be a lot of people. On the other hand, I think my original PR (I haven't checked this one) also failed on unknown keys, and a bunch of people are currently emitting the License PEP metadata that hasn't landed... I think hatch does that? 

My hope is for this to be as strict as we can though, so I can stick it in the upload flow to warehouse and stop the bleeding :)